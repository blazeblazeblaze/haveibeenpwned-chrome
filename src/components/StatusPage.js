import React from 'react';
import PropTypes from 'prop-types';

import ResultPage  from './ResultPage';
import EmailForm from './EmailForm';

const StatusPage = ({email, isPwned, onSubmit, onStartOver}) => {
  return(
    <div>
      {email && <ResultPage email={email} isPwned={isPwned} onStartOver={onStartOver} />}
      {!email && <EmailForm onSubmit={onSubmit} />}
    </div>
  )
}

StatusPage.propTypes = {
  isPwned: PropTypes.bool,
  email: PropTypes.string,
  onSubmit: PropTypes.func.isRequired,
  onStartOver: PropTypes.func.isRequired
};

export default StatusPage;
