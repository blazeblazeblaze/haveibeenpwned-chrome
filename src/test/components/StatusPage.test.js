import React from 'react';
import ReactDOM from 'react-dom';
import StatusPage from '../../components/StatusPage';
import renderer from 'react-test-renderer';

jest.mock('../../components/ResultPage', () => () => 'ResultPage');
jest.mock('../../components/EmailForm', () => () => 'EmailForm');

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<StatusPage onStartOver={jest.fn} onSubmit={jest.fn} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('StatusPage renders EmailForm when email is set', () => {
  const email = 'foo@example.com';

  const component = renderer.create(
    <StatusPage onStartOver={jest.fn} onSubmit={jest.fn} email={email} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('StatusPage renders ResultPage when email is not set', () => {
  const component = renderer.create(
    <StatusPage onStartOver={jest.fn} onSubmit={jest.fn} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
