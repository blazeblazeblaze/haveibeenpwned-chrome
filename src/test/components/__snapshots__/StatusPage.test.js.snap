// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusPage renders EmailForm when email is set 1`] = `
<div>
  ResultPage
</div>
`;

exports[`StatusPage renders ResultPage when email is not set 1`] = `
<div>
  EmailForm
</div>
`;
